sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -4442,39 +4442,63 @@ private void SetNameFieldsIfNull()
                     nameof(BlobBaseClient),
                     message:
                     $"{nameof(Uri)}: {Uri}");
-
-                string operationName = $"{nameof(BlobBaseClient)}.{nameof(Exists)}";
-
+                DiagnosticScope scope = ClientConfiguration.ClientDiagnostics.CreateScope($"{nameof(BlobBaseClient)}.{nameof(Exists)}");
                 try
                 {
-                    Response<BlobProperties> response = await GetPropertiesInternal(
-                        conditions: default,
-                        async: async,
-                        cancellationToken: cancellationToken,
-                        operationName)
-                        .ConfigureAwait(false);
+                    scope.Start();
 
-                    return Response.FromValue(true, response.GetRawResponse());
-                }
-                catch (RequestFailedException storageRequestFailedException)
-                when (storageRequestFailedException.ErrorCode == BlobErrorCode.BlobNotFound
-                    || storageRequestFailedException.ErrorCode == BlobErrorCode.ContainerNotFound)
-                {
-                    return Response.FromValue(false, default);
-                }
-                catch (RequestFailedException storageRequestFailedException)
-                when (storageRequestFailedException.ErrorCode == BlobErrorCode.BlobUsesCustomerSpecifiedEncryption)
-                {
-                    return Response.FromValue(true, default);
+                    // Get the blob properties, but don't throw any exceptions
+                    RequestContext context = new()
+                    {
+                        ErrorOptions = ErrorOptions.NoThrow,
```

Comment:
This will suppress an exception from being thrown from within GetProperties, but error logging and tracing will still occur in the pipeline.  In order to suppress those as well, you'll need to add classifiers using context.AddClassifier(...)

Reply:
Sigh...  That's what I get for rushing and only checking the Storage tests.  I updated and verified that distributed tracing is clean too.

Current:
![Current](https://user-images.githubusercontent.com/1179329/159102927-7d25608a-2dbe-4c9f-aa03-cdbf18a6da20.png)

Fixed:
![Fixed](https://user-images.githubusercontent.com/1179329/159102930-ca774b2b-7e02-45c0-ae33-78e587243323.png)

```C#
#define Fixed

using Azure;
using Azure.Core;
using Azure.Monitor.OpenTelemetry.Exporter;
using Azure.Storage.Blobs;
using Azure.Storage.Blobs.Models;
using Azure.Storage.Blobs.Specialized;
using OpenTelemetry.Trace;
using System.Diagnostics;

// Settings
const string StorageConnStr = "...";
const string AppInsightsConnStr = "...";
const string Name =
#if !Fixed
    "Current";
#else
    "Fixed";
#endif

// Collect telemetry in Azure Monitor
AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
using TracerProvider telemetry =
    OpenTelemetry.Sdk
    .CreateTracerProviderBuilder()
    .AddSource(Name, "Azure.*")
    .AddAzureMonitorTraceExporter(options => options.ConnectionString = AppInsightsConnStr)
    .Build();
using ActivitySource source = new(Name);
using Activity? group = source.StartActivity(Name)!;


// Check if a fake blob exists
BlobClient fake = new BlobClient(StorageConnStr, "fakecontainer", "fakeblob");
Console.WriteLine($"Exists: {fake.Exists().Value}");


// Try acquiring a lease from a fake blob
#if !Fixed

try
{
    BlobLease lease = fake.GetBlobLeaseClient().Acquire(TimeSpan.FromSeconds(15));
    Console.WriteLine($"Lease: {lease.LeaseId}");
}
catch (RequestFailedException ex)
{
    Console.WriteLine($"Failed to obtain lease!\n{ex}");
}

#else

RequestContext context =
    new RequestContext() { ErrorOptions = ErrorOptions.NoThrow }
    .AddClassifier(404, BlobErrorCode.BlobNotFound.ToString(), false)
    .AddClassifier(404, BlobErrorCode.ContainerNotFound.ToString(), false)
    .AddClassifier(409, BlobErrorCode.LeaseAlreadyPresent.ToString(), false);
Response<BlobLease> response =
    fake.GetBlobLeaseClient().Acquire(
        TimeSpan.FromSeconds(15),
        conditions: null,
        context);
Response raw = response.GetRawResponse();
if (raw.Status == 201)
{
    Console.WriteLine($"Lease: {response.Value.LeaseId}");
}
else
{
    Console.WriteLine($"Failed to obtain lease!\n{raw.Content.ToString()}");
}

#endif
```